clients/src/main/java/org/apache/kafka/clients/tools/VerifiableProducer.java

```diff
@@ -233,6 +273,11 @@ String successString(RecordMetadata recordMetadata, String key, String value, Lo
     
         public void onCompletion(RecordMetadata recordMetadata, Exception e) {
             synchronized (System.out) {
+                if (VerifiableProducer.this.stopProducing.get()) {
```

Comment:
Not sure about this behavior. The whole point of this producer is to log complete information about what it sent and what it knows about the success of those sends. If we ignore some callbacks, we won't have logged all the data. It seems very likely we would regularly see "extra" data at consumers if we used this approach.

One solution would be to still log these events, but adjust the output when you see an error and `stopProducing` is true, e.g. use `producer_send_closing_error` instead of `producer_send_error` to indicate it may have just been due to a timeout (although again, with the addition of KIP-19's patch with client-side timeouts, we may have to rethink what exactly these messages mean since sends could succeed but timeout too quickly).